app/javascript/hooks/queries/admin/roles/useRole.jsx

```diff
@@ -0,0 +1,10 @@
+import { useQuery } from 'react-query';
+import axios from '../../../../helpers/Axios';
+
+export default function useRole(roleId) {
+  return useQuery(
+    ['getRole', roleId],
+    () => axios.get(`/admin/roles/${roleId}.json`).then((resp) => resp.data.data),
+    { retry: 0 },
```

Comment:
Why `retry: 0`?

Reply:
React queries by default retries the query on error 3 times.
There's no hope or need to wait for 3 retrials to reroute the user back to `roles` page.
The first error is adequate otherwise they user will have to wait for the retrials delay.

Reply:
This isn't something we can just randomly introduce in 1 PR - these kinds of things need to be applied across all mutations. 

Also, there could be an instant when the DB goes down for a milisecond causing the first request to fail. A second request could go through successfully

Reply:
Alright, let's address it in all queries in a future PR.